Products/Zuul/catalog/tests/test_model_catalog.py

```diff
@@ -0,0 +1,169 @@
+import unittest
+import transaction
+import random
+
+from Products.Zuul.tests.base import BaseTestCase, init_modelcatalog
+from Products.Zuul.catalog.model_catalog import ModelCatalogClient
+from Products.Zuul.catalog.interfaces import IModelCatalog
+from Products.ZenModel.Device import Device
+from Products.ZenModel.ZDeviceLoader import JobDeviceLoader
+from Products.ZenUtils.ZenScriptBase import ZenScriptBase
+from zenoss.modelindex.model_index import SearchParams
+from zope.component import getUtility
+
+dmd = ZenScriptBase(connect=True).dmd
```

Comment:
also, here https://github.com/zenoss/zenoss-prodbin/blob/develop/Products/ZenTestCase/BaseTestCase.py#L149 we make sure that tests don't commit. Are your calls to commit working? if so we may be introducing undesired behavior in the test framework